libvast/src/segment.cpp

```diff
@@ -90,31 +84,14 @@ chunk_ptr segment::chunk() const {
 caf::expected<std::vector<table_slice>>
 segment::lookup(const vast::ids& xs) const {
   std::vector<table_slice> result;
-  auto f = [](auto buffer) noexcept {
-    return visit(
-      detail::overload{
-        [](const fbs::table_slice::legacy::v0* slice) noexcept {
-          return std::pair{slice->offset(), slice->offset() + slice->rows()};
-        },
-      },
-      buffer->data_nested_root());
+  auto f = [&](auto flat_slice) noexcept {
+    auto slice = table_slice{*flat_slice, chunk_, table_slice::verify::no};
+    return std::pair{slice.offset(), slice.offset() + slice.rows()};
```

Comment:
We should add a TODO for storing a separate offset table in the segment header. Doing it like here causes lots of small disk reads.